File: 日常实验/实验3/task3.py
import cv2
lena = cv2.imread("./lena.jpg")
horizontal_mirror_lena = cv2.flip(lena, 0)
cv2.imshow("原始图像", lena)
cv2.imshow("水平镜像翻转图像", horizontal_mirror_lena)
cv2.waitKey()
cv2.destroyAllWindows()

File: 日常实验/实验4/task3.m
lena = imread('lena.jpg');
lena_gray=rgb2gray(lena);
subplot(2,2,1);
imshow(lena_gray);
title('原始图像');


LEN1=12;
THREN1=22;
PFS2=fspecial("motion",LEN1,THREN1);
fuzzy_lena=imfilter(lena_gray,PFS2,'circular'); %对图像进行模糊劣化
subplot(2,2,2);
imshow(fuzzy_lena);
title('模糊劣化图像');


deconvwnr_lena=deconvwnr(fuzzy_lena,PFS2);
subplot(2,2,3);
imshow(deconvwnr_lena);
title('采用维纳滤波复原图像');

least_squares_lena=deconvreg(fuzzy_lena,PFS2);
subplot(2,2,4);
imshow(least_squares_lena);
title('采用最小二乘法复原图像')